r2-opds/src/main/java/org/readium/r2/opds/OPDS1Parser.kt

```diff
@@ -45,13 +49,43 @@ object Namespaces {
 class OPDS1Parser {
     companion object {
 
+        suspend fun parseUrlString(url: String): Try<ParseData, Exception> {
```

Comment:
It could be interesting to take a `client` argument to let the app provide its own implementation if necessary.

Also an additional `parseRequest()` API taking directly an `HttpRequest` would let the app customize the HTTP headers, timeouts and other settings without a custom `HttpClient`. This way we don't need to provide `parseUrlString(url: String, headers: MutableMap<String, String>)`.

```suggestion
        suspend fun parseUrlString(url: String, client: HttpClient = DefaultHttpClient()): Try<ParseData, Exception> {
        ...
        }
        
        suspend fun parseRequest(request: HttpRequest, client: HttpClient = DefaultHttpClient()) ...
```

Reply:
I think we can actually just have one function in that case:  Accepts a URL as a string, and accepts an HttpClient with the default being DefaultHttpClient.  And `parseRequest` sounds better than `parseUrlString`.